datafusion/physical-expr/src/array_expressions.rs

```diff
@@ -228,10 +228,10 @@ fn compute_array_dims(arr: Option<ArrayRef>) -> Result<Option<Vec<Option<u64>>>>
 
 fn check_datatypes(name: &str, args: &[&ArrayRef]) -> Result<()> {
     let data_type = args[0].data_type();
-    if !args
-        .iter()
-        .all(|arg| arg.data_type().equals_datatype(data_type))
-    {
+    if !args.iter().all(|arg| {
+        arg.data_type().equals_datatype(data_type)
+            || arg.data_type().equals_datatype(&DataType::Null)
```

Comment:
@Veeupup Why do we need Null checking here, which function failed without this?

Reply:
when it comes to handle multi arrays functions,their input types maybe like `null, List(Int32)...`. Then, but we should handle `null` type in function body rather than return error here.